### PR TITLE
[test]: add more watchdog test cases.

### PR DESCRIPTION
**Pull request by XuGuohui**
As the title describes.

### Completeness

- [x] User is totes amazing for contributing!
- [x] Contributor has signed CLA ([Info here](https://github.com/spark/firmware/blob/develop/CONTRIBUTING.md))
- [ ] Problem and Solution clearly stated
- [ ] Run unit/integration/application tests on device
- [ ] Added documentation
- [ ] Added to CHANGELOG.md after merging (add links to docs and issues)

### CODE REVIEW

File: user/tests/wiring/watchdog/watchdog.cpp [hunk 1]
```diff
@@ -55,8 +68,117 @@ test(WATCHDOG_02_default) {
...
+    assertEqual(0, pushMailbox(MailboxEntry().type(MailboxEntry::Type::RESET_PENDING), 5000));
+    System.sleep(SystemSleepConfiguration().mode(SystemSleepMode::STOP).duration(10s));
+}
+
+test(WATCHDOG_06_default_running_in_none_hibernate_mode_2) {
+    assertTrue(System.resetReason() == RESET_REASON_WATCHDOG);
+    checkState(WatchdogState::DISABLED);
+}
+
+static retained uint32_t magick = 0;
```
Comment (avtolstoy) [inline user/tests/wiring/watchdog/watchdog.cpp:127]: Are retained variables kept both on Gen 3 and P2/Tracker M after WDG reset?

**Review by avtolstoy — APPROVED** [review]

File: user/tests/wiring/watchdog/watchdog.cpp (hunk 1 above)
Comment (XuGuohui) [reply user/tests/wiring/watchdog/watchdog.cpp:127]: Not on P2 based platforms

Comment (avtolstoy) [reply user/tests/wiring/watchdog/watchdog.cpp:127]: There is `hal_backup_ram_sync()`, so in theory you can perform that before WDG reset and have backup ram restored on boot.

Comment (XuGuohui) [reply user/tests/wiring/watchdog/watchdog.cpp:127]: We stilll need to figure out how we can promise the `retained` key word works as expected implicitly.

Comment (avtolstoy) [reply user/tests/wiring/watchdog/watchdog.cpp:127]: debug peripheral watchpoint for that memory region on writes?

Comment (XuGuohui) [reply user/tests/wiring/watchdog/watchdog.cpp:127]: I'll create a SC ticket for tracking this problem.

**Pull request merged**